src/spikeinterface/core/unitsaggregationsorting.py

```diff
@@ -77,20 +77,32 @@ def __init__(self, sorting_list, renamed_unit_ids=None):
             if np.all(annotations == annotations[0]):
                 self.set_annotation(annotation_name, sorting_list[0].get_annotation(annotation_name))
 
+        # Check if all the sortings properties have the same dtype
         property_keys = {}
         property_dict = {}
         deleted_keys = []
+        np_dtypes = [np.integer, np.floating, np.str_, bool]
```

Comment:
Do we have any concern that numpy could change again. np.bool is gone I believe np.str_ has changed (although I think the type itself is still consistent). Just curious--> this has caused problems on phy which had to be updated to account for changes in numpy types over time.

Reply:
Yeah, I'm a bit worried too. And numpy doesn't really have a set of "basic super types". Hmmm. One option is to try and refactor so that all these checks are controlled from one place? Then at least fixing bugs will be easier??? Very difficult!

Reply:
The risk of that is we then end up making a string parser. This is a tricky problem. Is there a way we just force this to always be bigger strings? It takes a memory hit, but then we don't worry about the dtype. But I don't know str memory representation to know if trying to just pick a big value for the dtype is truly a waste.

Reply:
```suggestion
        np_dtypes = [np.integer, np.floating, np.str_, np.bool]
```

Before numpy 2.0 this is an alias for bool and after that it should work.

Reply:
One thing to keep in mind, but I'm fuzzy on... I believe there is a long running "bug" in NumPy that happens depending on the length of the array when combining strings and ints such that the behavior is unexpected. On their issue thread they basically said they haven't touched it because the group is split about what is the right way to handle the issue. @h-mayorquin do you remember this issue with trying to merge sortings with some one sorting having str ids and one having int ids?

Reply:
Hello, just writing down what we discussed outside of GitHub: we agreed to use `dtype.kind` and allow numpy to try a concatenation if the types are the same. This means we don't have to do any special string checks.